docs/current_docs/manuals/developer/functions.mdx

```diff
@@ -180,26 +291,31 @@ The result will look like this:
 Hello, world
 ```
 
-:::note
-The API considers any type that accepts a `None` value to be optional. Thus,
-if the `name` argument is omitted from an API request, it will be set to `None`
-automatically.
-
-However, when used directly from Python code, it's in fact a required argument
-since it has no default value. In this context, the term "optional" is the same
-concept as the older [`typing.Optional[X]`](https://docs.python.org/3/library/typing.html#typing.Optional)
-annotation for a type that allows the `None` value.
-:::
```

Comment:
Ok to remove as this general section will need to be generally improved after resolving:
- https://github.com/dagger/dagger/issues/6749

We'll want to focus more on optionals being the ones with default values. We've normalized the SDKs to separate _nullable_ from _has default_, but null values aren't really used by the API, so this needs to change.